### ISSUE TITLE
codegen: Use direct increase instead of map update for count function

### ISSUE THREAD

**Issue by danobi** [iovisor/bpftrace#2782]
From @olsajiri :
```
So the code generated for count follows this logic:

  val = map_lookup(key)
  if (!val)
    val = 0;
  val += 1
  map_update(key, val)

The idea of the change is to skip map_update call and do the
increase directly in the memory, like:

  ptr = map_lookup(key)
  if (!ptr)
    map_update(key, 1)
  else
   *ptr += 1
```

Initial work is in https://github.com/iovisor/bpftrace/pull/1300 . I think it should be relatively straightforward to pick this up and try `CreateAdd()` (see PR discussion).

**Comment by jordalgo** [comment]
Working on this here: https://github.com/iovisor/bpftrace/pull/2795

**Comment by viktormalik** [comment]
Fixed by #2795.
